tests/unit/configFile.spec.js

```diff
@@ -133,4 +133,239 @@ describe("configFile.js", () => {
+      const defaultProvidersGroup = [
+        { name: "IP Lookup", enabled: true },
+        { name: "Domain", enabled: true },
+        { name: "Hash", enabled: false },
+      ];
+
+      // If mergeGroups is false shouldn't override.
+      settings.mergeGroups = false;
+      await LocalStore.setOne(StoreKey.SETTINGS, settings);
+      await ConfigFile.default.parseJSONFile(newData);
+      settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      expect(settings.providersGroups).toEqual(defaultProvidersGroup);
+
+      // If mergeGroups is false should override providersGroups.
+      defaultProvidersGroup[2].name = "New Hash";
+      newData.groups[2][1] = "New Hash";
+      settings.mergeGroups = true;
+      await LocalStore.setOne(StoreKey.SETTINGS, settings);
+      await ConfigFile.default.parseJSONFile(newData);
+      settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      expect(settings.providersGroups).toEqual(defaultProvidersGroup);
+    });
+
+    it("search providers are merged, overridden and ignored depending on corresponding flag", async () => {
+      var settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      var serachProviders = await LocalStore.getOne(StoreKey.SEARCH_PROVIDERS);
+
+      // By default flag should be merge and duplicate values shouldn't be added.
+      var newData = {
+        searchproviders: [
+          [
+            -1,
+            "Google new",
+            "http://www.google2.com/search?q=TESTSEARCH",
+            true,
+            true,
+            3,
+            false,
+            "",
+            false,
+            "",
+          ],
+          [
+            -1,
+            "D - AlienVault OTX Domain",
+            "https://otx.alienvault.com/indicator/domain/TESTSEARCH",
+            true,
+            true,
+            2,
+            false,
+            "",
+            false,
+            "",
+          ],
+        ],
+      };
+
+      await ConfigFile.default.parseJSONFile(newData);
+
+      serachProviders.push({
+        menuIndex: -1,
+        label: "Google new",
+        link: "https://www.google2.com/search?q=TESTSEARCH",
+        enabled: true,
+        fromConfig: true,
+        group: 3,
+        postEnabled: false,
+        postValue: "",
+        proxyEnabled: false,
+        proxyUrl: "",
+      });
+      var newSerachProviders = await LocalStore.getOne(
+        StoreKey.SEARCH_PROVIDERS
+      );
+      expect(newSerachProviders).toEqual(serachProviders);
+
+      // Override flag should change searchProviders to new values.
+      settings.mergeSearchProviders = "override";
+      await LocalStore.setOne(StoreKey.SETTINGS, settings);
+      await ConfigFile.default.parseJSONFile(newData);
+      newSerachProviders = await LocalStore.getOne(StoreKey.SEARCH_PROVIDERS);
+      expect(newSerachProviders).toEqual(
+        ConfigFile.default.parseProviders(newData.searchproviders)
+      );
+    });
+  });
```

Comment:
Please split this test case into three tests: "providers are merged (and duplicates aren't added)", "providers are overridden" and "providers are ignored".